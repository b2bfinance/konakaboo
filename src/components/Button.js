import styled, { css } from 'styled-components';

const Button = styled.a`
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  font-weight: ${props => props.theme.mainBoldFontWeight};
  justify-content: center;
  padding: 0.75rem 1.5rem;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  user-select: none;

  ${props =>
    props.primary &&
    css`
      background-color: ${props.theme.applyButtonBackground};
      color: ${props.theme.applyButtonColor};
    `};

  ${props =>
    props.secondary &&
    css`
      background-color: ${props.theme.infoButtonBackground};
      color: ${props.theme.infoButtonColor};
    `};

  ${props =>
    props.slim &&
    css`
      padding: 0.25rem 0.5rem;
      font-size: 90%;
    `};
`;

export default Button;
