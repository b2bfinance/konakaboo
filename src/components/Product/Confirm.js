import React from 'react';
import styled from 'styled-components';
import Button from '../Button';
import Row from '../Row';
import Col from '../Col';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';

export const ConfirmButtonCol = styled(Col)`
  margin: 1.5rem auto 0;
`;

export const ConfirmDialog = styled(Dialog)`
  text-align: center;
`;

const ProductConfirm = ({
  open,
  handleRequestClose,
  title,
  description,
  forwardUrl
}) => (
  <ConfirmDialog open={open} onClose={handleRequestClose}>
    <DialogTitle>{title}</DialogTitle>
    <DialogContent>
      <DialogContentText>{description}</DialogContentText>
      <Row>
        <ConfirmButtonCol phone={80} tablet={60} desktop={40}>
          <Button primary href={forwardUrl}>
            Apply
          </Button>
        </ConfirmButtonCol>
      </Row>
    </DialogContent>
  </ConfirmDialog>
);

export default ProductConfirm;
