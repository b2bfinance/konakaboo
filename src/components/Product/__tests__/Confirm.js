import React from 'react';
import { shallow, render } from 'enzyme';
import Button from '../../Button';
import Confirm, {
  handleConfirmClick,
  ConfirmButtonCol,
  ConfirmDialog
} from '../Confirm';

test('<Confirm /> renders correctly', () => {
  const {
    heading,
    description
  } = stubData.products.withConfirmation.meta.confirm;

  expect(
    shallow(<Confirm title={heading} description={description} />)
  ).toMatchSnapshot();
});

test('<ConfirmButtonCol /> renders correctly', () => {
  expect(render(<ConfirmButtonCol />)).toMatchSnapshot();
});

test('<ConfirmDialog /> renders correctly', () => {
  expect(shallow(<ConfirmDialog />)).toMatchSnapshot();
});
