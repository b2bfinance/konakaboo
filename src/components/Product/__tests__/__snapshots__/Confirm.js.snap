// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirm /> renders correctly 1`] = `
<Styled(WithStyles)>
  <WithStyles>
    Application via Brokers only
  </WithStyles>
  <WithStyles>
    <WithStyles>
      B2Bfinance.com are registered intermediaries who will quote, package and submit your lending request to Natwest. There is no charge for this service as we are paid a small commission by Natwest to cover our time.
    </WithStyles>
    <styled.div>
      <styled.div
        desktop={40}
        phone={80}
        tablet={60}
      >
        <styled.a
          primary={true}
        >
          Apply
        </styled.a>
      </styled.div>
    </styled.div>
  </WithStyles>
</Styled(WithStyles)>
`;

exports[`<ConfirmButtonCol /> renders correctly 1`] = `
.c0 {
  width: 100%;
  min-height: 1px;
  margin: 1.5rem auto 0;
}

<div
  class="c0"
>
  
</div>
`;

exports[`<ConfirmDialog /> renders correctly 1`] = `
.c0 {
  text-align: center;
}

<WithStyles
  className="c0"
/>
`;
